app/controllers/sales-orders/show.js

```diff
@@ -1,14 +1,22 @@
-import Ember from 'ember';
-import computed from 'ember-computed-decorators';
+import Ember from "ember";
+import computed from "ember-computed-decorators";
 
-const { filterBy, notEmpty } = Ember.computed;
+const { notEmpty } = Ember.computed;
 
 export default Ember.Controller.extend({
-  hasDataPath: notEmpty('dataPath'),
+  hasDataPath: notEmpty("dataPath"),
 
-  filteredItems: filterBy('items', 'isSold', true),
+  @computed("items.[]", "model.orderItems.[]")
```

Comment:
Anytime you use a property on one of the items in the collection, we should use the prop as a dependent key, otherwise we won't get updates when things change.

Example: you are filtering on `isSold`, if this happens to change, then we won't retrigger this function since it is not declared as dependent.

```js
@computed("items.[]", "model.orderItems.[]")
```

Should therefore be:

```js
@computed("items.@each.{isSold}", "model.orderItems.[]")
```

Or maybe we can use the helper:

```js
products: filterBy('items', 'isSold', true),

@computed("products.[]", "model.orderItems.[]")
unusedProducts(products, orderItems){
  return products
    .filter(p => 
      p.any(oi => oi.get("item.id") === p.get("id")));
}
```